src/test/logger.spec.ts

```diff
@@ -0,0 +1,98 @@
+import { IChildLogger } from "@vscode-logging/types";
+import { noopLogger } from "../noop-logger";
+import { IRpc } from "../rpc-common";
+import { RpcMock } from "./rpc-mock";
+
+const sum = (a: number, b: number): number => {
+  return a + b;
+};
+
+describe("Logger tests", () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("When logger is not passed noop is used", () => {
+    let rpc1: IRpc = new RpcMock();
+    let rpc2: IRpc = new RpcMock();
+    (rpc1 as RpcMock).setPeer(rpc2 as RpcMock);
+    (rpc2 as RpcMock).setPeer(rpc1 as RpcMock);
+    const trace = jest.spyOn(noopLogger, "trace");
+
+    rpc1.registerMethod({ func: sum });
+    const param1: number = 1;
+    const param2: number = 2;
+    rpc2.invoke("sum", [param1, param2]).then((value) => {
+      console.log(`sum is ${value}`);
+      expect(value).toBe(param1 + param2);
+    });
+    expect(trace).toHaveBeenCalledTimes(2);
+    expect(trace).toHaveBeenCalledWith(
+      expect.stringMatching(
+        /Handling request id: \d\.\d+ method: sum parameters: 1,2/
+      )
+    );
+    expect(trace).toHaveBeenCalledWith(
+      expect.stringMatching(
+        /Handling response for id: \d\.\d+ message success flag is: true/
+      )
+    );
+  });
+
+  it("When logger is passed it is used", () => {
```

Comment:
consider to split to 2-3 more specific unit test

Reply:
i test single flow here, rpc2 sends command to rpc1m which triggers error log,
dont think there is something to split here

Reply:
np